crates/ty_project/src/metadata.rs

```diff
@@ -27,6 +27,9 @@ pub struct ProjectMetadata {
     /// The raw options
     pub(super) options: Options,
 
+    /// Config file to override any discovered configuration
+    pub(super) config_file_override: Option<SystemPathBuf>,
```

Comment:
I'm not sure about this, but we could consider combining it with `extra_configuration_paths`, considering that they're exclusive to each other.